R/user-defined-pk-parameter.R

```diff
@@ -83,26 +55,24 @@ DynamicPKParameter <- R6::R6Class("DynamicPKParameter",
     #' @param name Name of the user defined PK-Parameter
     #' @param standardPKParameter What PK-Parameter should be used to perform calculation.
     #' See \code{StandardPKParameter} enum for all possible pk parameters
-    #' @return A new `DynamicPKParameter` object.
-    initialize = function(ref = NULL, name = NULL, standardPKParameter = NULL) {
-      validateIsString(name, nullAllowed = TRUE)
-      validateEnumValue(StandardPKParameter, standardPKParameter, nullAllowed = TRUE)
-      ref <- ref %||% rClr::clrNew("OSPSuite.Core.Domain.PKAnalyses.DynamicPKParameter")
+    #' @return A new `UserDefinedPKParameter` object.
+    initialize = function(name, standardPKParameter) {
+      validateIsString(name)
+      validateEnumValue(standardPKParameter, StandardPKParameter)
+      ref <- rClr::clrNew("OSPSuite.Core.Domain.PKAnalyses.UserDefinedPKParameter")
       super$initialize(ref)
-      # Because of weird issue with nullable value in rClr
-      if (!is.null(name)) {
-        self$name <- name
-      }
-      if (!is.null(standardPKParameter)) {
-        self$standardPKParameter <- standardPKParameter
-      }
+      self$name <- name
+      self$standardPKParameter <- standardPKParameter
     },
     #' @description
     #' Print the object to the console
     #' @param ... Rest arguments.
     print = function(...) {
       private$printClass()
       private$printLine("Name", self$name)
+      private$printLine("DisplayName", self$displayName)

```

Comment:
A PK Parameter (or user defined pk parameter which is a derived class) has a Name, DisplayName, Dimension and displayUnit
Values hould be exported in the table in the display unit